src/classes/services/CaseService.cls

```diff
@@ -0,0 +1,60 @@
+/**
+ * Created by Mariya on 25.01.2023.
+ */
+
+public inherited sharing class CaseService {
+
+    public static Boolean isOwnerChanged(Case newCase, Case oldCase) {
+        return newCase.OwnerId != oldCase.OwnerId;
+    }
+
+    public static Boolean isCaseStatusAssigned(Case caseRecord) {
+        return caseRecord.Status == CaseUtility.ASSIGNED_STATUS;
+    }
+
+    public static Boolean isAppointmentRequestRT(Case caseRecord) {
+        return caseRecord.RecordTypeId == CaseUtility.getAppointmentRequestRTId();
+    }
+
+    private static Map<Id, Case> getCasesWithNewOwnerByOwnerIds(List<Case> newCases, Map<Id, Case> oldCasesByIds) {
+
+        Map<Id, Case> filteredCasesByOwnerIds = new Map<Id, Case>();
+
+        for (Case updatedCase : newCases) {
+            if (isOwnerChanged(updatedCase, oldCasesByIds.get(updatedCase.Id)) && isAppointmentRequestRT(updatedCase)) {
+                filteredCasesByOwnerIds.put(updatedCase.OwnerId, updatedCase);
+            }
+        }
+
+        return filteredCasesByOwnerIds;
+    }
+
+    public static void createAppointmentResources(List<Case> newCases, Map<Id, Case> oldCasesByIds) {
+
+        Map<Id, Case> filteredCasesByOwnerIds = getCasesWithNewOwnerByOwnerIds(newCases, oldCasesByIds);
+
+        List<Appointment_Resource__c> appointmentResources = new List<Appointment_Resource__c>();
+
+        Map<Id, Boutique_Resource__c> boutiqueResourcesByUserIds = new Map<Id, Boutique_Resource__c>();
+
+        for (Boutique_Resource__c boutiqueResource : BoutiqueResourceSelector.getBoutiqueResourceByUserIds(filteredCasesByOwnerIds.keySet())) {
+            if (BoutiqueResourceService.isTeamMemberRT(boutiqueResource)) {
+                boutiqueResourcesByUserIds.put(boutiqueResource.User__c, boutiqueResource);
+            }
+        }
+
+        for (Id caseOwnerId : filteredCasesByOwnerIds.keySet()) {
+            if (boutiqueResourcesByUserIds.containsKey(caseOwnerId)) {
```

Comment:
MINOR
This condition is not needed here, as there could not be such case when there is no record.